src/include/simeng/OS/Process.hh

```diff
@@ -148,18 +148,26 @@ class Process {
   /** Unique pointer to FileDescArray class.*/
   std::unique_ptr<FileDescArray> fdArray_;
 
-  // Thread state
-  // TODO: Support multiple threads per process
-  /** The clear_child_tid value. */
-  uint64_t clearChildTid = 0;
-
   /** Current status of the process. */
   procStatus status_ = procStatus::waiting;
 
   /** The CPU context associated with this process. Used to enable context
    * switching between multiple processes. */
   cpuContext context_;
 
+  /** Holds the address that the process should write its TID to.
```

Comment:
Can it please be reworded to say: The memory address at which the process writes its TID. The default value is 0 and is only updated when the process is created using the clone syscall with the CLONE_CHILD_SETTID flag; upon creation, the first thing the thread should do is store/write its TID at this address.

Reply:
Given its use case can the variable not be named `childTid_`? Default it either to UINT64_MAX or 0. If it is set to a value other than default then write it to memory. 
Upon reading the syscall spec I think its more similar to actual impl

Reply:
Process and thread are used interchangeably either this should be clarified or the difference between them should be explicitly defined.

Reply:
Renaming to `childTid_` could become confusing when we get on to support parent and child processes (i.e. for MPI). 

In the documentation it is named `set_child_tid` and so I think our implementation should reflect that in order to maintain consistency. And it is already defaulted to 0.

Reply:
Which documentation is being refereed here? I looked at 
![image](https://user-images.githubusercontent.com/62028695/220891077-c49285b9-4fb6-446a-b96e-82d46266bbc4.png)
Apologies If i've looked at the wrong one

Reply:
Looks like some inconsistency with the docs. I have been looking at https://man7.org/linux/man-pages/man2/set_tid_address.2.html

Reply:
this is the clone syscall docs

Reply:
As discussed offline - I think until it is exactly clear the distinction / overlap between them it is better to be more verbose. If it turns out during the clone syscall that both are not needed then I will happily change it to your suggestion